File: medinex_frontend/src/app/services/pregunta.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable, catchError } from 'rxjs';
import { throwError } from 'rxjs';
import baserUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class PreguntaService {

  constructor(private http:HttpClient) { }

  public listarPreguntasDelDoctor(doctorId:any): Observable<any>{
    const url = `${baserUrl}/pregunta/doctor/todos/${doctorId}`;
    console.log('🔗 PreguntaService - URL completa del endpoint:', url);
    
    return this.http.get(url).pipe(
      catchError((error) => {
        console.error('❌ PreguntaService - Error:', error);
        console.error('❌ PreguntaService - URL que falló:', url);
        return throwError(() => error);
      })
    );
  }

  public guardarPregunta(pregunta:any){
    return this.http.post(`${baserUrl}/pregunta/`,pregunta);
  }

  public eliminarPregunta(preguntaId:any){
    return this.http.delete(`${baserUrl}/pregunta/${preguntaId}`);
  }

  public actualizarPregunta(pregunta:any){
    return this.http.put(`${baserUrl}/pregunta/`,pregunta);
  }

  public obtenerPregunta(preguntaId:any){
    return this.http.get(`${baserUrl}/pregunta/${preguntaId}`);
  }

  public listarPreguntasDelDoctorParaLaPrueba(doctorId:any){
    return this.http.get(`${baserUrl}/pregunta/doctor/todos/${doctorId}`);
  }

  public evaluarDoctor(preguntas:any){
    return this.http.post(`${baserUrl}/pregunta/evaluar-doctor`,preguntas);
  }
}

File: medinex-backend/src/main/java/com/sistema/rep/servicios/PreguntaService.java
package com.sistema.rep.servicios;

import com.sistema.rep.modelo.Doctor;
import com.sistema.rep.modelo.Pregunta;

import java.util.Set;

public interface PreguntaService {

    Pregunta agregarPregunta(Pregunta pregunta);

    Pregunta actualizarPregunta(Pregunta pregunta);

    Set<Pregunta> obtenerPreguntas();

    Pregunta obtenerPregunta(Long preguntaId);

    Set<Pregunta> obtenerPreguntasDelDoctor(Doctor doctor);

    void eliminarPregunta(Long preguntaId);

    Pregunta listarPregunta(Long preguntaId);

}
